File: NERP.Admin.Web1/ClientApp/src/app/civilworks/models/civilwork.ts
import { DbBaseEntity } from '../../common/models/dbBaseEntity';
import { CustomField } from '../../common/models/customField';
import { FileAttachment } from '../../common/models/FileAttachment';
import { Supplier } from '../../common/models/supplier';

export class Civilwork extends DbBaseEntity {
    name: string;
    code?: string;
    parentId?: string;
    parentEntityType?: string;
    customFields?: CustomField[] = new Array<CustomField>();
    photos?: FileAttachment;
    attachments?: FileAttachment;
    suppliers?: Supplier[] = new Array<Supplier>();
}

File: NERP.NetworkEntities/Models/Civilwork.cs
using Common.DataAccess.Mongo;
using NERP.Common.Models;
using System.Collections.Generic;

namespace NERP.NetworkEntities.Models
{
    public class Civilwork : DbBaseEntity
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public string Icon { get; set; }
        public string ParentId { get; set; }
        public string ParentEntityType { get; set; }
        public List<FileAttachment> Photos { get; set; }
        public List<FileAttachment> Attachments { get; set; }
        public List<Supplier> Suppliers { get; set; }

    }
}
